pkg/domain/infra/tunnel/containers.go

```diff
@@ -972,7 +972,7 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 	// Wait for all events to be read
 	mutex.Lock()
 	if eventsErr != nil || lastEvent == nil {
-		logrus.Errorf("Cannot get exit code: %v", err)
+		logrus.Errorf("Cannot get exit code: %v, last event: %v", eventsErr, lastEvent)
```

Comment:
The new message doesn't make much sense, now you are just logging
```
"Cannot get exit code: <nil>, last event: <nil>"
```
in case where the last event is nil and no error

In general it is not clear how do you reproduce this? Overall we made many changes to how Wait() works recently so I really do not see any point of trying reading events when wait failed so IMO it would be much better to get rid of this event code instead of trying to fix.

Reply:
There is no real error, the condition is `||` which means we log when lastEvent == nil even when there was no error. What happens here is that we can read events but it will simply not contain our wanted exited events as such event the event is nil.

What podman version are you using? Any chance you can test with podman from main?

Reply:
We tried latest podman (5.2.2), still seeing the same error.

Reply:
> podman (5.2.2)

This has not most of the wait fixes, you need to wait for 5.3 or use podman from main.